### PR TITLE
[WIP] ppc64le

### PR DESCRIPTION
**Pull request by clnperez**
Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by clnperez** [comment]
I'm running through these and just getting everything to build atm. Doug did some testing but I don't think he did the whole docker runtime test. I'm working on that this week and next.

File: mk/common.mk [hunk 1]
```diff
@@ -15,12 +15,12 @@ INSTALL  ?= install
...
 OBJCPY   ?= objcopy
 RPCGEN   ?= rpcgen
-BMAKE    ?= MAKEFLAGS= bmake
+BMAKE    ?= MAKEFLAGS= make 
 
 DATE     := $(shell date -u --iso-8601=minutes)
 REVISION := $(shell git rev-parse HEAD)
 COMPILER := $(realpath $(shell which $(CC)))
-PLATFORM ?= $(shell uname -p)
+PLATFORM ?= $(shell uname -m)
```
Comment (clnperez) [inline mk/common.mk]: uname -p for debian returns "unknown", in case you're wondering why I changed this.

File: mk/common.mk [hunk 2]
```diff
@@ -38,8 +38,6 @@ getdef = $(shell sed -n "0,/$(1)/s/\#define\s\+$(1)\s\+\(\w*\)/\1/p" $(2))
 
 ifeq ($(PLATFORM),x86_64)
 getarch = $(shell [ -f /etc/debian_version ] && echo "amd64" || echo "x86_64")
-else ifeq ($(PLATFORM),ppc64le)
-getarch = $(shell [ -f /etc/debian_version ] && echo "ppc64el" || echo "ppc64le")
-else
-$(error Unsupported architecture)
+else ifeq (,$(filter $(PLATFORM),ppc64le ppc64el))
```
Comment (clnperez) [inline mk/common.mk]: i think i need to undo this, actually. i was trying to figure out what was going on with the debian build.

**Comment by clnperez** [comment]
the package builds work for centos and ubuntu, but for debian I'm running into:

```
cd /mnt/libnvidia-container_1.0.0_ppc64el && debuild clean                                                                                                  
This package has a Debian revision number but there does not seem to be                                                                                     
an appropriate original tar file or .orig directory in the parent directory;                                                                                
(expected one of libnvidia-container_1.0.0~alpha.2.orig.tar.gz, libnvidia-container_1.0.0~alpha.2.orig.tar.bz2,                                             libnvidia-container_1.0.0~alpha.2.orig.tar.lzma,  libnvidia-container_1.0.0~alpha.2.orig.tar.xz or libnvidia-container_1.0.0_ppc64el.orig)                  
continue anyway? (y/n) Use of uninitialized value $ans in pattern match (m//) at /usr/bin/debuild line 951.                                                 Makefile:290: recipe for target 'deb' failed                                                                                                                
make: *** [deb] Error 1
```

**Comment by 3XX0** [comment]
Thanks! I was already working on unifying the build process so I only took few bits from this PR, I hope you don't mind.
It should build fine on PPC64 now, for example the following gives you the latest release for Xenial:
`make docker-ubuntu:16.04 WITH_LIBELF=yes TAG=alpha.3 DISTRIB=xenial`

**Comment by clnperez** [comment]
Not at all. That's great news @3XX0. I'll try this out!

**Comment by clnperez** [comment]
@3XX0 I built a stretch, centos7, and the ubuntu 16.04 packages. ✔️ ✔️ ✔️

**Pull request closed without merging**